fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1647,6 +1648,22 @@ public LoanTermVariationsData mapRow(final ResultSet rs, @SuppressWarnings("unus
 
     }
 
+    @Override
+    public Collection<LoanScheduleDelinquencyData> retrieveScheduleDelinquencyData(LocalDate businessLocalDate) {
+        LoanScheduleDelinquencyMapper mapper = new LoanScheduleDelinquencyMapper(DateUtils.getBusinessLocalDate());
+        final StringBuilder sqlBuilder = new StringBuilder(400);
+        sqlBuilder.append("select ").append(mapper.schema())
```

Comment:
Could you please explain why we are relying on the native query here instead of a regular JPQL/Criteria API query?

Reply:
Done!